warehouse/query-core/src/main/java/datawave/query/util/regex/RegexTrie.java

```diff
@@ -0,0 +1,159 @@
+package datawave.query.util.regex;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * This class can be used to turn a large list of strings into a regex.
+ * This may be used to handle queries that arrive with lists so large that the term limit is reached.
+ */
+public class RegexTrie {
+    final Node root = new Node();
+    class Node {
+        final HashMap<Character, Node> next = new HashMap<>();
+        boolean terminator = false;
+    }
+
+    public RegexTrie() {
+    }
+
+    public RegexTrie(List<String> strings) {
+        addAll(strings);
+    }
+
```

Comment:
Not sure where else this could be used. Is it worth adding a `RegexTrie(String... strings)` constructor too?

Reply:
that would be reasonable.  will do